extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -361,13 +363,32 @@ public void handle(Void e) {
             });
         }
         // Headers sent on any request, regardless of the response
-        Map<String, String> headers = httpConfiguration.header;
+        Map<String, HeaderConfig> headers = httpConfiguration.header;
         if (!headers.isEmpty()) {
-            httpRouteRouter.route().order(Integer.MIN_VALUE).handler(new Handler<RoutingContext>() {
-                @Override
-                public void handle(RoutingContext event) {
-                    event.response().headers().addAll(headers);
-                    event.next();
+            // Creates a handler for each header entry
+            headers.forEach((name, config) -> {
```

Comment:
Let's avoid `forEach` when headers is empty. There is no reason to have the JVM initialize the lambda (which we should also probably avoid) for no reason.

Reply:
> Let's avoid `forEach` when headers is empty. 

The `forEach` is only reached when the headers map is not empty


> There is no reason to have the JVM initialize the lambda (which we should also probably avoid) for no reason.

I thought JDK 11+ had enhancements in the forEach, but sure, I can change that